src/escalus_ct.erl

```diff
@@ -92,7 +101,16 @@ do_log_stanza(Jid, Direction, Stanza) ->
                        [ReportString, Stanza]),
                 ct:fail(Error)
         end,
-    ct:pal(stanza_log, "~s~n~s", [ReportString, PrettyStanza]).
+    {PrintFun, PrintArgs} = case Target of
+                                console_and_file ->
+                                    {pal, [ReportString, PrettyStanza]};
+                                console ->
+                                    {print, [ReportString, PrettyStanza]};
+                                file ->
+                                    {log, [ReportString, exml_nif:escape_cdata(PrettyStanza)]}
```

Comment:
This way we get different file output with `console_and_file` and `file`. If we call `print` + `log` instead of `pal` for the `console_and_file` option, we could escape it so that it would be consistent.

Reply:
I'm not sure I understand.
I intentionally escaped the argument for `ct:log`, since it *was not* displayed correctly in the HTML files, while with the former `ct:pal` (before this PR) it was displayed correctly without any additional manual escaping.

Reply:
Indeed, `pal` automatically escapes the argument it passes to `log`: https://github.com/erlang/otp/blob/afb9501f99bf94b05f999cd341dade721f69d023/lib/common_test/src/ct_logs.erl#L592-L594

Reply:
Ok, I see, you are correct. It can be a bit misleading though, as it confused me when reading the code.

Reply:
I've rewritten it to be more explicit. It uses the CT API now, so no manual escaping is necessary.